### PR TITLE
Login with Github functionality

### PR DESCRIPTION
**Pull request by dn1470**
GithubURL.ts file added 
- Client_id: Currently using dummy data in the page but Dyno oauth app's client id and pass word will be stored in AWS parameter store at next pull request
- Current Scope: user:email but needs to be discussed

Login.tsx 
- User is directed to GitHub login page when they click login button

### CODE REVIEW

File: frontend/src/client/models/GithubURL.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export function GitHubURL(from: string) {
```
Comment (The-TT-Hacker) [inline frontend/src/client/models/GithubURL.ts]: the src/client directory is autogenerated. dont add anything to it manually

File: frontend/src/client/models/GithubURL.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  const rootUrl = 'https://github.com/login/oauth/authorize';
+  const options = {
+    client_id: 'c2f25de6e7cfc7713944' as string,
```
Comment (The-TT-Hacker) [inline frontend/src/client/models/GithubURL.ts]: put this stuff in the settings.json as well. its generated by terraform and then you can store these variables in terraform variables

File: frontend/src/client/models/GithubURL.ts (hunk 1 above)
Comment (dn1470) [reply frontend/src/client/models/GithubURL.ts]: should i then just create a separate directory within "pages" -> "login" or just move the GithubURL.ts file to login directory?

Comment (The-TT-Hacker) [reply frontend/src/client/models/GithubURL.ts]: you can probably include all of this in the Login.tsx file

File: frontend/src/client/models/GithubURL.ts (hunk 2 above)
Comment (The-TT-Hacker) [reply frontend/src/client/models/GithubURL.ts]: https://github.com/stevensnicholas/comp9447-team2/blob/main/frontend/public/settings.json

File: frontend/src/pages/login/Login.tsx [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+type LocationState = {
+  from: {
+    path: string;
+  };
+};
+
+function GitHubURL(from: string) {
+  const rootUrl = 'https://github.com/login/oauth/authorize';
+  const options = {
+    client_id: '' as string,
```
Comment (The-TT-Hacker) [inline frontend/src/pages/login/Login.tsx]: get the settings from App.tsx

Comment (The-TT-Hacker) [reply frontend/src/pages/login/Login.tsx]: and pass the client_id in as a prop

**Comment by The-TT-Hacker** [comment]
not sure what you've done but you now have a ton of unwanted changes. please rebase main

**Comment by The-TT-Hacker** [comment]
@thefishua why did you add all of those commits to the PR? its a mess now

**Comment by dn1470** [comment]
got messy during rebasing. should i just close this pr and start a new branch? I get stuck at rebasing

**Comment by The-TT-Hacker** [comment]
rebasing will always be a pain when you are in the habit of piling on tons of commits instead of resetting and committing again. usually there should only be 1 commit per PR but this one has 40+. you either just have to go through all 40 commits during the rebase or make a new branch from main and re-implement your changes.

**Pull request closed without merging**